teams/infra/pull-request/automerge/action.yml

```diff
@@ -0,0 +1,27 @@
+---
+name: Automerge Pull Requests
+description: |
+  Automatically merge pull requests.
+  The action fetches all open pull requests with a specific label (default: `automerge`) and merges them.
+  The merge method is set to squash to allow a requirement for signed commits on the base branch.
+  At the moment, this action does not work with the `workflow_dispatch` event due to some flaws in the third-party action.
+
+inputs:
+  github-token:
+    description: 'Github token with permissions to modify pull requests, contents and actions'
+    required: true
+    secret: true
+  label:
+    description: 'Label to filter pull requests'
+    required: false
+    default: 'automerge'
+
+runs:
+  using: composite
+  steps:
+  - name: automerge
+    uses: pascalgn/automerge-action@v0.16.4
+    env:
+      GITHUB_TOKEN: "${{ inputs.github-token }}"
+      MERGE_METHOD: "squash"
```

Comment:
(non-blocking)
what was the reasoning for going with `squash` vs `merge` and preserving the git history?

Reply:
It was a coincidence that I discovered it:
If a repositories branch protection is set to "require signed commits" then "merge" doesn't work anymore since a merge commit seems to remove the signature, whereas a `squash` seems to work just fine. At least on my testing repository. We can also go with "merge" or "rebase", but I also like the default squash message which refers to the release PR.